Choose monospaced font in C++ code rather in `*.ui` file

hebasto opened the pull request:
This change makes macOS choose the correct monospaced font.

**fanquake**  noted (https://github.com/bitcoin/bitcoin/pull/16432#issuecomment-514486077):
> the monospace font looks a bit weird on macOS

... because it is _not_ monospaced.

This is an alternative to #79 as that PR has a Concept NACK from [**luke-jr**](https://github.com/bitcoin-core/gui/pull/79#issuecomment-686510806).

I'm personally still lean to embedding font as it makes the GUI more stable.

hebasto commented:
cc @fanquake @luke-jr

hebasto commented:
@jonasschnelli Should this be considered a bugfix?

jonasschnelli commented:
Unsure about this.
IMO this is a MVC violation (which we have plenty but unsure to add more).

hebasto commented:
> IMO this is a MVC violation (which we have plenty but unsure to add more).

How is it so? I think both `qt/overviewpage.cpp` and `qt/forms/overviewpage.ui` are parts of "View" in MVC, no?

jonasschnelli commented:
> How is it so? I think both qt/overviewpage.cpp and qt/forms/overviewpage.ui belong to "View" in MVC, no?

I'd say *.ui is view, *.cpp is controller (the non-automatic generated ones).

hebasto commented:
> > How is it so? I think both qt/overviewpage.cpp and qt/forms/overviewpage.ui belong to "View" in MVC, no?
> 
> I'd say *.ui is view, *.cpp is controller (the non-automatic generated ones).

Right. But Qt programming usually assumes "the view and the controller objects are combined" (https://doc.qt.io/qt-5/model-view-programming.html).

luke-jr commented on src/qt/overviewpage.cpp:
```diff
@@ -141,6 +141,18 @@ OverviewPage::OverviewPage(const PlatformStyle *platformStyle, QWidget *parent)
+
+    QFont f = GUIUtil::fixedPitchFont();
+    f.setWeight(QFont::Bold);
+    f.setPointSize(10);
```
Let's not override this for no reason?

hebasto replied:
The only reason is to make things explicitly rather than implicitly. Does `setPointSize` call has any downsides?

luke-jr replied:
Yes, the downside is that it is explicit and overrides the user's preference (which may not be 10pt).

hebasto replied:
Thanks! [Updated](https://github.com/bitcoin-core/gui/pull/87#issuecomment-716163108).

Now I'm able to change the font size with the `qt5ct` utility on Linux Mint 20 :)

luke-jr reviewed (changes requested):
Looks good to me, one change requested

hebasto commented:
Updated a44ba74ceccf6f61c65972e2ae37327de0db4c79 -> 2e386cd3dd3fea5a22dc8ae871b80a80f8cdb3f7 ([pr87.01](https://github.com/hebasto/gui/commits/pr87.01) -> [pr87.02](https://github.com/hebasto/gui/commits/pr87.02), [diff](https://github.com/hebasto/gui/compare/pr87.01..pr87.02)):

- addressed @luke-jr's [comment](https://github.com/bitcoin-core/gui/pull/87#discussion_r511600141):
> Yes, the downside is that it is explicit and overrides the user's preference (which may not be 10pt).

luke-jr reviewed (approved):
utACK

jarolrod reviewed (approved):
Tested ACK on macOS 10.15.7

Master
<img width="427" alt="master" src="https://user-images.githubusercontent.com/23396902/98738345-8ede2100-2375-11eb-8bfc-ef4a78e3e15c.png">

PR
<img width="427" alt="pr" src="https://user-images.githubusercontent.com/23396902/98738354-9271a800-2375-11eb-95c8-03027c1fde5c.png">

jonasschnelli commented:
Master:
<img width="1297" alt="Bildschirmfoto 2021-01-28 um 10 26 02" src="https://user-images.githubusercontent.com/178464/106117406-44d63180-6153-11eb-99df-c7ce56525e4d.png">

This PR:
<img width="1285" alt="Bildschirmfoto 2021-01-28 um 10 23 57" src="https://user-images.githubusercontent.com/178464/106117429-4a337c00-6153-11eb-8b81-090d0251943b.png">


I think the correct monospace font should be bigger and eventually bold.

jonasschnelli commented:
Adding `f.setPointSize(12);` makes it look much better (on macOS, unsure about windows/linux):

hebasto commented:
> Adding `f.setPointSize(12);` makes it look much better (on macOS, unsure about windows/linux):

Right. But @luke-jr's opinion is [opposite](https://github.com/bitcoin-core/gui/pull/87#discussion_r511600141).

RandyMcMillan commented:
It would be great to add a font size selector - similar to the console interface. 

![Screen Shot 2021-01-28 at 10 07 51 PM](https://user-images.githubusercontent.com/152159/106226484-56502580-61b5-11eb-9137-ef74af03645b.png)

hebasto commented:
Closed in favor of #207.
